src/sorunlib/commands.py

```diff
@@ -2,26 +2,60 @@
 import datetime as dt
 
 
-def wait(target_time):
+def wait_until(timestamp, tolerance=None):
     """Wait until a specified time.
 
     Args:
-        target_time (str): Time in ISO format and in UTC timezone to wait
+        timestamp (str): Time in ISO format and in UTC timezone to wait
             until, i.e. "2015-10-21T07:28:00", "2023-01-01T0:00:00+00:00"
+        tolerance (int or str): Tolerance on the difference between the time
+            when the function is evaluated and `timestamp`. Can be specified as
+            a ISO formatted timestamp (str) or in number of seconds (int).
+
+            When evaluated, if `tolerance` (str) is in the past or if the
+            difference between the current time and `timestamp` is greater than
+            this `tolerance` (int), then an error is raised. If `None`, an
+            error will not be raised. Default is `None`.
+
+    Raises:
+        ValueError: If `timestamp` has an unsupported timezone, `tolerance`
+            is an unsupported type, or if the current time at evaluation is past
+            the threshold set by the `tolerance`.
 
     """
-    target = dt.datetime.fromisoformat(target_time)
-    TZ = target.tzinfo
+    target = dt.datetime.fromisoformat(timestamp)
 
+    # Determine timezone
+    TZ = target.tzinfo
     if TZ not in [None, dt.timezone.utc]:
         offset = target.tzinfo.tzname(None)
         raise ValueError(f'Unsupported timezone ({offset}) detected. '
                          + 'Timezone must be UTC.')
 
+    # Grab current TZ aware timestamp
     now = dt.datetime.now(TZ)
 
-    assert target > now, f"time {target} is in the past"
-
-    diff = target - now
-    print(f"Waiting for {diff.total_seconds()} seconds")
-    time.sleep(diff.total_seconds())
+    # Determine "deadline" from tolerance
+    if tolerance is None:
+        deadline = None
+    elif isinstance(tolerance, int):
+        deadline = target + dt.timedelta(seconds=tolerance)
+    elif isinstance(tolerance, str):
+        deadline = dt.datetime.fromisoformat(tolerance)
```

Comment:
You don't check the timezone of tolerance.  That leads to this unexpected TypeError:
```
>>> wait_until('2023-01-01T00:00', '2023-01-01T00:00+00:00')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "commands.py", line 51, in wait_until
    elif now > deadline:
TypeError: can't compare offset-naive and offset-aware datetimes
```
See my general review comments tho...

Reply:
Fixed when responding to general comments in [91029bb](https://github.com/simonsobs/sorunlib/pull/61/commits/91029bb0a9ccd874374cabf37fb546f882e98853).